jaeger-micrometer/src/main/java/com/uber/jaeger/micrometer/MicrometerStatsReporter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016, Uber Technologies, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.uber.jaeger.micrometer;
+
+import static io.micrometer.core.instrument.Metrics.counter;
+import static io.micrometer.core.instrument.Metrics.gauge;
+import static io.micrometer.core.instrument.Metrics.timer;
+
+import com.uber.jaeger.metrics.StatsReporter;
+import io.micrometer.core.instrument.ImmutableTag;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Tag;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MicrometerStatsReporter implements StatsReporter {
```

Comment:
StatsReporter should be deprecated, because it's very inefficient. StatsFactory is the preferred method (which we may also want to deprecate and rename to MetricsFactory to be consistent with Go)

Reply:
#338 created for this

Reply:
Sorry, my point (besides deprecating the StatsXXX API) was that this class should implement StatsFactory, not StatsReporter. The main Metrics class uses the factory to create stats objects once and call methods on them. In contrast, if this implements the StatsReporter you have to do very expensive `translateTags(tags)` for every `incCounter`, making it orders of magnitude more expensive call.

Reply:
I see what you mean now. I'll deprecate the StatsReporter as part of #338, but I changed this commit to get it to implement `StatsFactory`